### ISSUE TITLE
Sync behaviour

### ISSUE THREAD

**Issue by gwutz** [huseyz/obsidian-wallabag#21]
Hi @huseyz ! I didn't found a Wiki for that project therefore I post my thoughts about that wonderful plugin here in order to get your point of view. First of all, as you already saw in my MR about the annotations, I have a specific workflow in mind in order to process information personally. I have set up a Wallabag server and this acts as my kitchen sink for reading and highlighting/annotate informations in my daily life. After that I want to process these informations/highlights into my obisidan vault as a long-living note. My requirements look more or less like:

- Sync regularly the annotations into my vault with a link to my wallabag article (this is already possible today)
- Remove a note when I don't need the annotation/article anymore. This would be possible today too, as your plugin keeps an index of synced articles.
- Sometimes it happens that I didn't process/read the article yet. Unfortunately these articles are maybe already synced and will stay synced forever.
- Sometimes it happens that I added additonal annotations (because maybe I re-read that article). These will be not re-synced either.

Thinking about these (personal) requirements maybe it makes sense to change a bit the sync behaviour. What I have in mind is: always re-sync and overwrite the current wallabag note except when the note was synced before and is now deleted and when the note contains a special marker tag which blocks rewriting for the future. 

What do you think about that?

Things, I didn't consider here: different workflows/styles with PDF annotations maybe. I see that the plugin also supports this and I didn't used it yet so maybe my plan above is flawed in this regard.

**Comment by huseyz** [comment]
Hi @gwutz,

Most of the plugin functionality is built around my own workflow, which is
1) Add articles to Wallabag
2) Sync to Obsidian
3) Read and take notes (I mostly annotate exported pdfs) in the created Obsidian note. 
Hence, the notes were designed to be immutable.

I'm happy to accommodate any other workflows such as yours with the condition that they're introduced as options, e.g., by having a setting to enable updating the existing notes.

**Comment by peterrus** [comment]
Just want to add that my workflow is very similar to @gwutz's so I welcome this feature. I notice there is a clear divide between the two workflows in this issue:

A. One annotates (optionally) in Wallabag and then downloads/updates locally to markdown/HTML.
B. One downloads pdf's (once) and annotates locally.

It might be interesting to clearly delineate between the settings (in the settings screen) that apply to workflow A and the ones that apply to workflow B. Maybe the user will have to choose either workflow A or B and then only see the relevant settings, or we could give users the option to even use both.

Just some thoughts.

**Comment by quantumgardener** [comment]
Sorry, but I'm a little confused with what the plugin does. Are annotations synced? I don't see them in my notes and I'm wondering if I'm missing a setting.

**Comment by huseyz** [comment]
@quantumgardener annotations are technically synced, but it's not documented. Use the template variable https://github.com/huseyz/obsidian-wallabag/blob/180d6c542f5611d515bbce2222848dff64759a34/src/note/NoteTemplate.ts#L29

Maybe give it a try and see whether it fits your use.

**Comment by quantumgardener** [comment]
That works. Thanks.

**Comment by jason-mehmel** [comment]
> @quantumgardener annotations are technically synced, but it's not documented. Use the template variable
> 
> https://github.com/huseyz/obsidian-wallabag/blob/180d6c542f5611d515bbce2222848dff64759a34/src/note/NoteTemplate.ts#L29
> 
> Maybe give it a try and see whether it fits your use.

Might be a dumb question: do we need to build our own templates if we're going to get this annotation feature in our imports from Wallabag? (As in, importing the annotations from Wallabag into the article import.)

**Comment by huseyz** [comment]
Yes, if you need any custom behavior, you need to create a template for yourself.

**Comment by jason-mehmel** [comment]
Do those templates go in the same place as any other obsidian templates, or
is there a specific wallabag template location?


~Jason
pronouns: he / him / his
***@***.*** | www.jasonmehmel.com

**Comment by quantumgardener** [comment]
The template can go anywhere you like in your vault. 

Set the "Article note template file" in the Wallabag settings.

**Comment by jason-mehmel** [comment]
> That works. Thanks.

Hey @quantumgardener would you be able to share the template you made for annotations? (I'm still a newb with Obsidian and I'm having a hard time figuring it out!

**Comment by quantumgardener** [comment]
Sure. Here you go.

```
---
tags:
  "{ tags }": 
datetime:
  "{ created_at }": 
updated: 
title:
  "{ article_title }": 
author:
  - 
publication-year: 
recommender: 
id: "{{id}}"
wallabag-link:
  "{ wallabag_link }": 
---
## Annotations
{{annotations}}

## Content
{{content}}
```

**Comment by jason-mehmel** [comment]
@quantumgardener Thank you!

I tweaked it a bit for myself, but it's working great!

Question: your template above has quote marks and only one set of curly brackets. Is that part of your obsidian setup somehow? I had to remove the quotes and add curly brackets for my own setup.

Here's what I have:

(Authors is currently nil because I don't think that is set up via the API yet. Are you adding 'publication year' and recommender on your own? I've left it blank on my end but kept it in case I decide to start using it!)


```
---
tags: {{tags}} 
datetime: {{created_at}}
updated: {{updated_at}}
title: {{article_title}}
author: {{authors}}
publication-year:
recommender:
id: {{id}}
wallabag-link: {{wallabag_link}}
Original-link: {{original_link}}
---

```

**Comment by quantumgardener** [comment]
Can't recall why the quote marks are there. For something like a title, it is so ' in the title can be accommodated without error.

Authors I'm probably adding manually. I have a Templater script that conforms article titles to a specific style and layout. It will fill in publication year from the date if not there. Recommender is used for anyone who recommends a book or article to me. So, some of this template is also because I treat the article as a source so it has to have the same fields as other sources like books.

**Comment by quantumgardener** [comment]
I see what happened now @jason-mehmel as I had to fix it myself. Something mucked up my template and converted {{ to "{

I suggest that you make title "{{article_title}}" with quotes or you might find titles with ' or " in them fail to be recognised as properties.

**Comment by jason-mehmel** [comment]
That's a good tip! I wonder if it would also help with those titles that have a colon in them, for example like _Lord of the Rings: Why it's a good book._ 

Right now, that colon is messing up the properties data, and I'm editing the Wallabag article to remove them. But maybe those quote marks around the title would also fix the problem without tweaking each title.

**Comment by quantumgardener** [comment]
Yes. Any punctuation. Question marks as well. On 19 Dec 2024, at 7:07 am, jason-mehmel ***@***.***> wrote:﻿
That's a good tip! I wonder if it would also help with those titles that have a colon in them, for example like Lord of the Rings: Why it's a good book.
Right now, that colon is messing up the properties data, and I'm editing the Wallabag article to remove them. But maybe those quote marks around the title would also fix the problem without tweaking each title.
